File: flutter_hr_2/flutter_hr/lib/models/attendance_record.dart
class AttendanceRecord {
  final int? id;
  final int employeeId;
  final String date;
  final String? clockInTime;
  final String? clockOutTime;
  final double? workHours;
  final String status;
  final String? remarks;

  AttendanceRecord({
    this.id,
    required this.employeeId,
    required this.date,
    this.clockInTime,
    this.clockOutTime,
    this.workHours,
    required this.status,
    this.remarks,
  });

  factory AttendanceRecord.fromJson(Map<String, dynamic> json) {
    return AttendanceRecord(
      id: json['id'],
      employeeId: json['employeeId'],
      date: json['date'],
      clockInTime: json['clockInTime'],
      clockOutTime: json['clockOutTime'],
      workHours: json['workHours']?.toDouble(),
      status: json['status'],
      remarks: json['remarks'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'employeeId': employeeId,
      'date': date,
      'clockInTime': clockInTime,
      'clockOutTime': clockOutTime,
      'workHours': workHours,
      'status': status,
      'remarks': remarks,
    };
  }
}

File: flutter_hr/HR-And-Payroll-Management-System-Backend/src/main/java/Israt/example/Hr/and/payroll/management/system/backend/entity/AttendanceRecord.java
package Israt.example.Hr.and.payroll.management.system.backend.entity;

import Israt.example.Hr.and.payroll.management.system.backend.enums.AttendanceStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "attendances")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AttendanceRecord {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "employee_id", nullable = false)
    private Long employeeId;

    @Column(nullable = false)
    private LocalDate date;

    @Column(name = "clock_in_time")
    private LocalDateTime clockInTime;

    @Column(name = "clock_out_time")
    private LocalDateTime clockOutTime;

    @Column(name = "work_hours")
    private Double workHours;

    @Enumerated(EnumType.STRING)
    @Column(length = 20, nullable = false)
    private AttendanceStatus status = AttendanceStatus.PRESENT;

    @Column(length = 1000)
    private String remarks;

    @Column(name = "is_overtime")
    private Boolean isOvertime = false;

    @Column(name = "overtime_hours")
    private Double overtimeHours;

    @Column(name = "overtime_reason", length = 1000)
    private String overtimeReason;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "updated_by")
    private User updatedBy;

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }
}
